src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py

```diff
@@ -0,0 +1,387 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from msrestazure.azure_exceptions import CloudError
+from azure.cli.core.commands.arm import parse_resource_id, is_valid_resource_id
+import azure.cli.core.azlogging as azlogging
+from azure.cli.core._util import CLIError
+from .custom import set_vm, _compute_client_factory
+logger = azlogging.get_az_logger(__name__)
+
+_DATA_VOLUME_TYPE = 'DATA'
+_STATUS_ENCRYPTED = 'Encrypted'
+
+extension_info = {
+    'Linux': {
+        'publisher': 'Microsoft.Azure.Security',
+        'name': 'AzureDiskEncryptionForLinux',
+        'version': '0.1'
+    },
+    'Windows': {
+        'publisher': 'Microsoft.Azure.Security',
+        'name': 'AzureDiskEncryption',
+        'version': '1.1'
+    }
+}
+
+
+def keyvault_mgmt_client_factory(**_):
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from azure.mgmt.keyvault import KeyVaultManagementClient
+    return get_mgmt_service_client(KeyVaultManagementClient)
+
+
+def enable(resource_group_name, vm_name,  # pylint: disable=too-many-arguments,too-many-locals, too-many-statements
+           aad_client_id,
+           disk_encryption_keyvault,
+           aad_client_secret=None, aad_client_cert_thumbprint=None,
+           key_encryption_keyvault_id=None,
+           key_encryption_key_url=None,
+           key_encryption_algorithm='RSA-OAEP',
+           volume_type=None):
+    '''
+    Enable disk encryption on OS disk, Data disks, or both
+    :param str aad_client_id: Client ID of AAD app with permissions to write secrets to KeyVault
+    :param str aad_client_secret: Client Secret of AAD app with permissions to
+    write secrets to KeyVault
+    :param str aad_client_cert_thumbprint: Thumbprint of AAD app certificate with permissions
+    to write secrets to KeyVaul
+    :param str disk_encryption_keyvault:KeyVault where generated encryption key will be placed to
+    :param str key_encryption_key_url:Versioned KeyVault URL of the KeyEncryptionKey used to
+    encrypt the disk encryption key
+    :param str key_encryption_keyvault_id: id of the KeyVault containing the key encryption key
```

Comment:
So if we need this keyvault to encrypt the disk encryption key, then we should also be able to specify the encryption key by name, no?  We shouldn't have to specify only a URI.